### PR TITLE
Implemented transformation of delimited value lists

### PR DESCRIPTION
**Pull request by pmenke-de**
Implements: #32

Basic implementation. Requests for changes welcome ;-)

### CODE REVIEW

File: src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.ui.DocumentAdapter;
+
+import shaded.org.apache.commons.text.StringEscapeUtils;
+
+/**
+ * @author Philipp Menke
+ */
+class DelimitedListDialog extends JDialog {
```
Comment (krasa) [inline src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java]: `extends JDialog` probably not needed

File: src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    private static final long                serialVersionUID = -9173380044265093635L;
+    private final        DelimitedListAction action;
+    private final        Editor              editor;
+
+    JPanel     contentPane;
+    JTextField tfDestDelimiter;
```
Comment (krasa) [inline src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java]: these type prefixes are ugly and do not belong to the Java world :)

File: src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    private JTextField   tfSourceDelimiter;
+    private JRadioButton rbSourceSelection;
+    private JRadioButton rbSourceClipboard;
+    private JCheckBox    cbAutoQuote;
+    private JTextField   tfQuote;
+    private JEditorPane  epPreview;
```
Comment (krasa) [inline src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java]: I have preview editor in TextAlignmentForm#myPreviewPanel, it is probably much nicer, check it out

Comment (krasa) [reply src/osmedile/intellij/stringmanip/transform/DelimitedListDialog.java]: And you will not need to limit the length

File: src/osmedile/intellij/stringmanip/transform/DelimitedListAction.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            @Override
+            protected void doExecute(@NotNull Editor editor, @Nullable Caret caret, DataContext dataContext) {
+                final Pair<Boolean, Settings> dialogResult = DelimitedListAction.this.showDialog(editor);
+                if (!dialogResult.first) {
+                    return;
+                }
+
+                new EditorWriteActionHandler() {
+                    @Override
+                    public void executeWriteAction(Editor editor, @Nullable Caret caret, DataContext dataContext) {
```
Comment (krasa) [inline src/osmedile/intellij/stringmanip/transform/DelimitedListAction.java]: add `MyApplicationComponent.setAction(DelimitedListAction.class);`

**Comment by pmenke-de** [comment]
Hi, sorry for the late reply. Haven't got much time in the last couple weeks.
I think, i've reworked all points you mentioned. Thanks for the suggestions :-)

**Comment by krasa** [comment]
Thanks!

Is it just me, or it the async stuff totally unreadable? :) I simplified it, check it out.
Also it seems that limiting is needed after all, Editor cannot handle millions of characters very well.

**Comment by krasa** [comment]
@pmenke-de if it should transform to/from CSV, then it need to be way smarter, for CSV containing:
```
"aaa","b,bb","ccc"
"aaa","b""bb","ccc"
```
https://www.mkyong.com/java/how-to-read-and-parse-csv-file-in-java/

It also does not handle the original requirement:
```
"Ford|Chevy|Honda|Toyota" --> ["Ford", "Chevy", "Honda", "Toyota"]
["Ford", "Chevy", "Honda", "Toyota"] --> "Ford|Chevy|Honda|Toyota"
```
Maybe it would be better to use some parser library, and choose between that and the current processing.

**Pull request merged**